polarCoordinateTraversalTest.m

```diff
@@ -43,18 +43,235 @@ function testRayEntersCircleAndGoesThroughOrigin(testCase)
+        expected_rVoxels     = [3,2,1];
+        expected_thetaVoxels = [2,2,2];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
+function testBeginAtCircleCenterPositiveXY(testCase)
+    min_bound = [0.0, 0.0];
+    max_bound = [30.0, 30.0];
+    ray_origin = [15.0, 15.0];
+    ray_direction = [2.0, 7.0];
+    circle_center = [15.0, 15.0];
+    circle_max_radius = 10.0;
+    num_radial_sections = 3;
+    num_angular_sections = 4;
+    t_begin = 0.0;
+    t_end = 20.0;
+    verbose = false;
+    
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [3,2,1];
+        expected_thetaVoxels = [0,0,0];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
 % Ray begins at a theta boundary (and goes in orthogonal directions).
-% Ray begins at a radial boundary (and does in orthogonal directions).
+function testBeginAtThetaBoundaryInOrthogonal(testCase)
+    min_bound = [0.0, 0.0];
+    max_bound = [30.0, 30.0];
+    ray_origin = [10.0, 15.0];
+    ray_direction = [0.0, 7.0];
+    circle_center = [15.0, 15.0];
+    circle_max_radius = 10.0;
+    num_radial_sections = 3;
+    num_angular_sections = 4;
+    t_begin = 0.0;
+    t_end = 20.0;
+    verbose = false;
+    
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [2,1];
+        expected_thetaVoxels = [1,1,1];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+% Ray begins at a radial boundary (and goes in orthogonal directions).
+function testBeginAtRadialBoundaryInOrthogonal(testCase)
+    min_bound = [0.0, 0.0];
+    max_bound = [30.0, 30.0];
+    ray_origin = [22.07, 22.07];
+    ray_direction = [-1.0, -1.0];
+    circle_center = [15.0, 15.0];
+    circle_max_radius = 10.0;
+    num_radial_sections = 3;
+    num_angular_sections = 4;
+    t_begin = 0.0;
+    t_end = 20.0;
+    verbose = false;
+    
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [1,2,3,3,2,1];
+        expected_thetaVoxels = [0,0,0,2,2,2];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
+% Special Test. Ray goes along a ray boundary
```

Comment:
Nit: I don't think "Special test" is necessary here. if you want to put something, then "Corner case" seems more appropriate.